clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubMessageHelper.java

```diff
@@ -58,4 +61,17 @@ public static String getTopicName(final String topic, final String prefix) {
         return String.format("%s.%s", prefix, topic);
     }
 
+    /**
+     * Gets the payload data contained in a Pub/Sub message's data body.
+     *
+     * @param message The Pub/Sub message data to parse.
+     * @return The bytes representing the payload or an empty byte array if the message neither has data.
+     *
+     * @throws NullPointerException if the message or data is {@code null}.
+     */
+    public static byte[] getPayload(final PubsubMessage message) {
+        Objects.requireNonNull(message);
+        Objects.requireNonNull(message.getData());
```

Comment:
FMPOV this method should not throw an exception if the message has no payload but instead return `null` or en empty Optional. In the end, it is not an exceptional case that a message has no payload, is it?

Reply:
Agree, since I found out that the Data can't be null (even if its not set, an empty byte array is set inside the client library) I updated it to return an empty optional if the Pubsubmessage is null or to return an optional of the byte array.

Reply:
In the end It is up to you, how you would like to implement this.
Based on what you found out, I guess I would do it like this:
* throw an NPE if the message is `null` (my understanding is that this will never happen because PubSub will always invoke your receiver with a non-null mPubsubMessage instance)
* return the payload from the message otherwise (which could then be an empty byte array)

The return type would then be `byte[]` as in your original approach ...

Reply:
updated as suggested